### DISCUSSION TITLE
If no pedals...

### DISCUSSION THREAD

**Discussion by larspalo** [GrandOrgue/grandorgue#153] [General]
Originally posted 2016-10-25 12:19:46.642000 by cemieux in GrandOrgue discussion forum on SourceForge<br><br>Hi,
Without pedals, how to configure a keyboard to play the lowest note of a chord? Feasible?

**Comment by larspalo** [comment]
Originally posted 2016-10-25 12:55:09.285000 by [Lars Palo](https://github.com/larspalo) in GrandOrgue discussion forum on SourceForge<br><br>Try a Bas coupler to pedal from any manual. See under Panels-&gt;Couplers when an organ is loaded.


 
      Från: cemieux &lt;cemieux@users.sf.net&gt;
 Till: [ourorgan:discussion] &lt;962124@discussion.ourorgan.p.re.sf.net&gt; 
 Skickat: tisdag, 25 oktober 2016 14:19
 Ämne: [ourorgan:discussion] If no pedals...
   
Hi,
Without pedals, how to configure a keyboard to play the lowest note of a chord? Feasible?If no pedals...Sent from sourceforge.net because you indicated interest in https://sourceforge.net/p/ourorgan/discussion/962124/To unsubscribe from further messages, please visit https://sourceforge.net/auth/subscriptions/

**Comment by larspalo** [comment]
Originally posted 2016-10-25 16:18:36.507000 by cemieux in GrandOrgue discussion forum on SourceForge<br><br>Yes, the use of &quot;Bass Couplerr&quot; with a manual works as I want!
Thanks !

cemieux
    Member
     
    Posts: 2
    Joined: Sat Apr 16, 2016 9:36 am

**Comment by mszbo** [comment]
Hello and thanks for all the hard work on this incredible software.

Is it possible to further develop this function, my problem is when playing legato, many times it fails to play couple the pedal bass. This happens when you press down multiple keys, and release the lowest note. The pedal note stops sounding until you release or press an additional low note again.
On HW and a johannus digital organ this isn't a problem. As soon as you release the lowest note, the next lowest note held down will sound the pedal pipes.

**Reply by larspalo** [reply]
A few years ago this was discussed in https://github.com/GrandOrgue/grandorgue/discussions/1425 and I came up with a slight improvement that you can test from https://github.com/larspalo/grandorgue/actions/runs/17532467443 (for some reason the OSX or mac builds are not working) and see if that improves the situation for you. However, if you'd read the whole discussion you'd find that it's not as simple as it might seem to decide the proper behavior of a BAS coupler for any particular keypress combination that it might initally seem.

That said, if you think my slight change to the algorithm is working for you and you'd like to have it implemented, please respond here and I might take the time to implement the choice that Oleg wanted before this could be implemented in the official GO.

**Reply by oleg68** [reply]
@larspalo may we add your enchancement with a special button that would switch between the old and the new coupler behavior?

**Reply by larspalo** [reply]
@oleg68 Yes, if I remember correctly we discussed a global option for selecting new or old BAS/MEL coupler implementation.
